argocd/datadog_checks/argocd/check.py

```diff
@@ -39,50 +42,25 @@ def __init__(self, name, init_config, instances):
 
     def parse_config(self):
-        self.scraper_configs = []
-        app_controller_endpoint = self.instance.get("app_controller_endpoint")
-        appset_controller_endpoint = self.instance.get("appset_controller_endpoint")
-        api_server_endpoint = self.instance.get("api_server_endpoint")
-        repo_server_endpoint = self.instance.get("repo_server_endpoint")
-        notifications_controller_endpoint = self.instance.get("notifications_controller_endpoint")
-
-        if (
-            not app_controller_endpoint
-            and not appset_controller_endpoint
-            and not repo_server_endpoint
-            and not api_server_endpoint
-            and not notifications_controller_endpoint
-        ):
-            raise ConfigurationError(
-                "Must specify at least one of the following:"
-                "`app_controller_endpoint`, `appset_controller_endpoint`, `repo_server_endpoint`, `api_server_endpoint`"
-                " or `notifications_controller_endpoint`."
-            )
+        endpoint_configs = [
+            ("app_controller_endpoint", APP_CONTROLLER_NAMESPACE, APPLICATION_CONTROLLER_METRICS),
+            ("appset_controller_endpoint", APPSET_CONTROLLER_NAMESPACE, APPSET_CONTROLLER_METRICS),
+            ("api_server_endpoint", API_SERVER_NAMESPACE, API_SERVER_METRICS),
+            ("repo_server_endpoint", REPO_SERVER_NAMESPACE, REPO_SERVER_METRICS),
+            ("notifications_controller_endpoint", NOTIFICATIONS_CONTROLLER_NAMESPACE, NOTIFICATIONS_CONTROLLER_METRICS),
+            ("commit_server_endpoint", COMMIT_SERVER_NAMESPACE, COMMIT_SERVER_METRICS),
+        ]
 
-        if app_controller_endpoint:
-            self.scraper_configs.append(
-                self.generate_config(app_controller_endpoint, APP_CONTROLLER_NAMESPACE, APPLICATION_CONTROLLER_METRICS)
-            )
-        if appset_controller_endpoint:
-            self.scraper_configs.append(
-                self.generate_config(appset_controller_endpoint, APPSET_CONTROLLER_NAMESPACE, APPSET_CONTROLLER_METRICS)
-            )
-        if api_server_endpoint:
-            self.scraper_configs.append(
-                self.generate_config(api_server_endpoint, API_SERVER_NAMESPACE, API_SERVER_METRICS)
-            )
-        if repo_server_endpoint:
-            self.scraper_configs.append(
-                self.generate_config(repo_server_endpoint, REPO_SERVER_NAMESPACE, REPO_SERVER_METRICS)
-            )
-        if notifications_controller_endpoint:
-            self.scraper_configs.append(
-                self.generate_config(
-                    notifications_controller_endpoint,
-                    NOTIFICATIONS_CONTROLLER_NAMESPACE,
-                    NOTIFICATIONS_CONTROLLER_METRICS,
-                )
-            )
+        self.scraper_configs = []
+        for endpoint_key, namespace, metrics in endpoint_configs:
+            endpoint = self.instance.get(endpoint_key)
+            if endpoint:
```

Comment:
nit: use the walrus operator to simplify it.

```suggestion
            if endpoint := self.instance.get(endpoint_key):
```